### PR TITLE
Fail on embedded provided deps

### PR DESCRIPTION
**Pull request by gjoranv**

### CODE REVIEW

File: config-model/pom.xml [hunk 1]
```diff
@@ -300,6 +300,11 @@
         <groupId>com.yahoo.vespa</groupId>
         <artifactId>bundle-plugin</artifactId>
         <extensions>true</extensions>
+        <configuration>
+          <allowEmbeddedArtifacts>com.fasterxml.jackson.core:jackson-annotations:2.15.2, com.fasterxml.jackson.core:jackson-core:2.15.2,
```
[the remark this answers is not in the document]
Comment (gjoranv) [reply config-model/pom.xml:304]: Including a different version of some artifacts could be a valid use case.

File: cloud-tenant-cd/pom.xml [hunk 2]
```diff
@@ -83,6 +83,8 @@
                 <artifactId>bundle-plugin</artifactId>
                 <extensions>true</extensions>
                 <configuration>
+                    <allowEmbeddedArtifacts>com.fasterxml.jackson.core:jackson-core:2.15.2, com.yahoo.vespa:annotations:8-SNAPSHOT,
+                        org.bouncycastle:bcpkix-jdk18on:1.74, org.bouncycastle:bcprov-jdk18on:1.74, org.bouncycastle:bcutil-jdk18on:1.74</allowEmbeddedArtifacts>
```
[the remark this answers is not in the document]
Comment (gjoranv) [reply cloud-tenant-cd/pom.xml:87]: I prefer that it's explicit. This should only be a temporary solution, or used in exceptional circumstances.

File: config-model/pom.xml [hunk 3]
```diff
@@ -301,9 +301,9 @@
         <artifactId>bundle-plugin</artifactId>
         <extensions>true</extensions>
         <configuration>
-          <allowEmbeddedArtifacts>com.fasterxml.jackson.core:jackson-annotations:2.15.2, com.fasterxml.jackson.core:jackson-core:2.15.2,
-            com.fasterxml.jackson.core:jackson-databind:2.15.2, com.yahoo.vespa:metrics:*, com.yahoo.vespa:predicate-search-core:*,
-            com.yahoo.vespa:searchcore:*</allowEmbeddedArtifacts>
+          <allowEmbeddedArtifacts>com.fasterxml.jackson.core:jackson-annotations, com.fasterxml.jackson.core:jackson-core,
```
Comment (bjorncs) [inline config-model/pom.xml:304]: These artifacts can likely be excluded here and re-embedded through model-fat.

**Review by bjorncs — APPROVED** [review]

File: config-model/pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (bjorncs) [reply config-model/pom.xml]: Agree assuming version is optional

File: config-model/pom.xml (hunk 3 above)
Comment (gjoranv) [reply config-model/pom.xml:304]: Yes, I'll create a follow-up ticket to review the exceptions.

**Pull request merged**